### PR TITLE
Add test utilities for clad::jacobian

### LINKED ISSUE #395: Add test utilities
Currently, we are having MACROs that makes writing tests easier in most of the test files. We should have unified testing utilities that can be included in any test file and used right away. 
With the upcoming support for differentiating class types and the associated tests, the need for generic and unified testing utilities is even more important. 

An example of testing utility interface for testing gradients may look like this: 

```cpp

INIT_GRADIENT(fn1);  // initialise gradient function for fn1 
INIT_GRADIENT(fn2, "i, j");  // initialise gradient function fn2 that computes derivatives with respect to parameters 'i' and 'j'

double a, d_a, d_b;
std::pair<double, long double> p, d_p;

// Resets derivative arguments to 0, executes the gradient function and then prints the derivatives.
TEST(fn1, /*NumOfDerivativeArgs=*/2, a, p, &d_a, &d_p); 
// Resets derivative arguments to 0, executes the gradient function and then prints the derivatives.
TEST(fn2, /*NumOfDerivativeArgs=*/3, a, 11, p, &d_a, &d_b, &d_p);
```

Currently, we need to add test utilities for the following differentiation modes, this list may soon increase if we decide to provide direct interface for pushforward and pullback functions.
- [x] Forward Mode
- [x] Reverse Mode
- [ ] Hessian Mode
- [ ] Jacobian Mode

### PR DESCRIPTION
**Pull request by Ris-Bali**
This PR aims to add test utilities for clad::jacobian References #395

### CODE REVIEW

**Review by parth-07 — COMMENTED** [review]
This is great. Can you please also add a test case which uses this test utility?

**Comment by Ris-Bali** [comment]
I've added an initial draft for now. Will add a TEST_JACOBIAN macro for CHECK-EXEC soon

File: test/Jacobian/testUtility.C [hunk 1]
```diff
@@ -0,0 +1,140 @@
...
+
+
+
+void fn_mul(double i, double j, double *res) {
+   res[0] = i*i;
+   res[1] = j*j;
+   res[2] = i*j;
+}
+
+void fn_mul_jac(double i, double j, double*res, double *jacobianMatrix);
```
Comment (parth-07) [inline test/Jacobian/testUtility.C:17]: Derivative functions declaration can be removed

File: test/TestUtils.h [hunk 2]
```diff
@@ -58,6 +58,50 @@ template <typename... Args> void display(Args... args) {
   printf("}\n");
 }
 
+void displayarr(double *arr, std::size_t n) {
```
[the remark this answers is not in the document]
Comment (Ris-Bali) [reply test/TestUtils.h]: Done !

**Review by parth-07 — COMMENTED** [review]
Please run clang-format in TestUtils.h and squash all the commits into one.

**Comment by parth-07** [comment]
Also, please change 'Fixes #395' to 'References #395' as one of the subitems in issue 395 is still remaining.

**Review by parth-07 — APPROVED** [review]
LGTM.

**Review by vgvassilev — APPROVED** [review]
Relying on @parth-07 here, good job!

**Pull request merged**